agenda/network/src/main/java/com/pronoidsoftware/agenda/network/KtorRemoteAgendaDataSource.kt

```diff
@@ -113,18 +128,96 @@ class KtorRemoteAgendaDataSource @Inject constructor(
         )
     }
 
+    // Events
+    override suspend fun createEvent(event: AgendaItem.Event): EmptyResult<DataError.Network> {
+        return httpClient.postMultipart(
+            route = AgendaRoutes.EVENT,
+            body = MultiPartFormDataContent(
+                formData {
+                    append(EVENT_CREATE_REQUEST, Json.encodeToString(event.toCreateEventRequest()))
+                    event.photos.forEachIndexed { index, photo ->
+                        val photoName = "photo$index"
+                        append(
+                            photoName,
+                            URL(photo.url).readBytes(),
+                            Headers.build {
+                                append(HttpHeaders.ContentType, "image/png")
+                                append(HttpHeaders.ContentDisposition, "filename=$photoName.png")
+                            },
+                        )
+                    }
+                },
+            ),
+        )
+    }
+
+    override suspend fun getEvent(id: String): Result<AgendaItem.Event, DataError.Network> {
+        return httpClient.get<EventDto>(
+            route = AgendaRoutes.EVENT,
+            queryParameters = mapOf(
+                EVENT_ID_QUERY_PARAM to id,
+            ),
+        ).map { it.toEvent(sessionStorage.get()?.userId) }
+    }
+
+    override suspend fun getAllEvents(): Result<List<AgendaItem.Event>, DataError.Network> {
+        return httpClient.get<AgendaDto>(
+            route = AgendaRoutes.FULL_AGENDA,
+        )
+            .map { agendaDto ->
+                agendaDto.events.map { it.toEvent(sessionStorage.get()?.userId) }
+            }
+    }
+
+    override suspend fun updateEvent(event: AgendaItem.Event): EmptyResult<DataError.Network> {
+        return httpClient.putMultipart(
+            route = AgendaRoutes.EVENT,
+            body = MultiPartFormDataContent(
+                formData {
+                    append(EVENT_CREATE_REQUEST, Json.encodeToString(event.toUpdateEventRequest()))
+                    event.photos.forEachIndexed { index, photo ->
```

Comment:
And for updating an event you'd want to filter this to only upload the local photos. Photos that are already saved on the backend and haven't been deleted don't have to be considered in this call.